File: simplypdf/src/main/java/com/wwdablu/soumya/simplypdf/composers/models/ImageProperties.java
package com.wwdablu.soumya.simplypdf.composers.models;

import com.google.gson.annotations.SerializedName;
import com.wwdablu.soumya.simplypdf.composers.ImageComposer;
import com.wwdablu.soumya.simplypdf.composers.UnitComposer;

public class ImageProperties extends UnitComposer.Properties {

    @SerializedName("imageurl")
    public String imageUrl;

    public UnitComposer.Alignment alignment;

    public ImageProperties() {
        alignment = UnitComposer.Alignment.LEFT;
    }

    @Override
    public String getPropId() {
        return ImageComposer.class.getName() + "Properties";
    }
}

File: simplypdf/src/main/java/com/wwdablu/soumya/simplypdf/composers/models/ImageProperties.kt
package com.wwdablu.soumya.simplypdf.composers.models

import com.google.gson.annotations.SerializedName
import com.wwdablu.soumya.simplypdf.composers.Composer
import com.wwdablu.soumya.simplypdf.composers.ImageComposer
import com.wwdablu.soumya.simplypdf.composers.UnitComposer

class ImageProperties : UnitComposer.Properties() {

    @SerializedName("source")
    @kotlin.jvm.JvmField
    var source: String? = null

    @SerializedName("format")
    @kotlin.jvm.JvmField
    var format: String? = null

    @kotlin.jvm.JvmField
    var alignment: Composer.Alignment = Composer.Alignment.START

    override val propId = ImageComposer::class.java.name + "Properties"
}